src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs

```diff
@@ -91,7 +138,7 @@ public void infers_from_list_types()
 
             type.Field(d => d.someList, nullable: true);
 
-            type.Fields.Last().Type.ShouldBe(typeof(ListGraphType<IntGraphType>));
+            type.Fields.Last().Type.ShouldBe(typeof(ListGraphType<NonNullGraphType<IntGraphType>>));
```

Comment:
It seems to me that the behavior regarding this property was wrong:
```c#
 public List<int> someList { get; set; }
```
Elements of this list can never have null value, so it should be coerced to `ListGraphType<NonNullGraphType<IntGraphType>>`.

I added tests for various combinations of types.